stats/src/main/java/io/airlift/stats/PauseMeter.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.stats;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import io.airlift.log.Logger;
+import io.airlift.units.Duration;
+import org.HdrHistogram.Histogram;
+import org.HdrHistogram.HistogramIterationValue;
+import org.weakref.jmx.Managed;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+
+public class PauseMeter
+{
+    private static final Logger LOG = Logger.get(PauseMeter.class);
+
+    private final long sleepNanos;
+
+    @GuardedBy("histogram")
+    private final Histogram histogram = new Histogram(3);
+
+    @GuardedBy("histogram")
+    private long totalPauseNanos;
+
+    private final Supplier<Histogram> snapshot = Suppliers.memoizeWithExpiration(this::makeSnapshot, 1, TimeUnit.SECONDS);
+
+    private final Thread thread;
+
+    // public to make it less likely for the VM to optimize it out
+    public volatile Long allocatedObject;
+
+    public PauseMeter()
```

Comment:
check how guice will work with these 2 constructors and whether we need to annotate one